src/ochubryk/hometask3/chocolate_bar.py

```diff
@@ -0,0 +1,148 @@
+# Шоколадка имеет вид прямоугольника, разделенного на n×m долек.
+# Шоколадку можно один раз разломить по прямой на две части.
+# Определите, можно ли таким образом отломить от шоколадки ровно k долек.
+# Определите, можно ли отломить от шоколадки ровно k одиночных долек.
+# Определите, можно ли отломить от шоколадки ровно k
+# одиночных долек с помощью t разломов
+
+
+def chocolate(n: int, m: int, k: int) -> int:
+    if k < (n * m):
+        if n == 1:
+            if k < m - 1:
+                return k
+        elif m == 1:
+            if k < n - 1:
+                return k
+        else:
+            if k % n == 0:
+                return k
+            elif k % n != n - 1:
+                return k + 1
+    elif k == (n * m):
+        return k - 1
+
+    return 0
+
+
+"""
+Шоколадку по условию можно один раз разломить на две части.
+Из этого следует, что:
+1) нельзя отломить  количество долек k меньше, чем m или n;
+2) нельзя отломить больше, чем общее количество долек m * n.
+"""
+# Количество долек k равно стороне m и равно стороне n
+# (k== m == n), общее количество долек(m*n) больше количества долек k
+
+assert chocolate(7, 7, 7) > 0, "True"
+
+# Количество долек k равно стороне n (k== n),
+# общее количество долек(m*n) больше количества долек k
+
+assert chocolate(6, 12, 6) > 0, "True"
+
+# Количество долек k не кратно ни стороне n, ни стороне m,
+# общее количество долек(m*n) больше количества долек k
+
+assert not chocolate(5, 15, 4) > 0, "False"
+
+# Количество долек k не кратно ни стороне n, ни стороне m,
+# общее количество долек(m*n) меньше количества долек k
+
+assert not chocolate(8, 8, 70) > 0, "False"
+"""
+При условии, что мы можем разломить шоколадку сколько угодно раз,
+мы получим k одиночных долек, если k < m*n -1
+"""
+
+
+assert chocolate(3, 2, 4) > 0, "True"
+assert chocolate(5, 4, 18) > 0, "True"
+assert not chocolate(2, 2, 3) > 0, "False"
+assert not chocolate(3, 2, 5) > 0, "False"
+"""
+При новом условии, мы можем сделать t разломов и
+должны получить ровно k одиночных долек.
+Во-первых, мы должны убедиться, что площадь шоколадки
+больше желаемого количества долек.
+Если количество одиночных долек, которое мы хотим получить(k),
+равно количеству всех долек в шоколадке k == (n * m),
+то количество разломов t == k - 1.
+Далее, мы имеем:
+1) количество одиночных долек, которое мы хотим получить (k)
+без остатка делится на сторону n: k % n == 0 -> t == k;
+2) количество одиночных долек, которое мы хотим получить (k)
+делится на сторону n с остатком, не равным n - 1:
+k % n != n - 1 -> t == k + 1;
+3) в случае, когда количество одиночных долек, которое мы хотим
+получить (k) делится на сторону n с остатком 1,
+получить необходимое количество долек k мы не сможем.
+Дополнительно рассмотрены случаи,
+когда одна из сторон шоколадки равна 1:
+1) n == 1 and k < m - 1, t == k;
```

Comment:
it's not all cases. i.e. we can cut k=5 single squares with 100 cuts for 1000x1000 chocolate. we'll just cut 5 of them and than will cut other 🍫  not taking single squares.

Reply:
There were comments about my previous solution. Then you added new conditions and all my functions were unified in one final function, but I forgot to delete these comments :)